quickwit/quickwit-janitor/src/actors/delete_task_planner.rs

```diff
@@ -236,8 +237,28 @@ impl DeleteTaskPlanner {
                         delete_query.end_timestamp,
                     );
                     // TODO: validate the query at the beginning and return an appropriate error.
-                    let tags_filter = extract_tags_from_query(&delete_query.query)
-                        .expect("Delete query must have been validated upfront.");
+                    let tags_filter = match &delete_query.query {
+                        Some(Query::Text(query_string)) => extract_tags_from_query(query_string)
+                            .expect("Delete query must have been validated upfront."),
```

Comment:
The TODO is right let's return an error. I am not 100% sure what upfront means here.

Reply:
that bit just got moved around, not sure what upfront means either